scheduler/src/cook/scheduler/constraints.clj

```diff
@@ -138,6 +148,49 @@
                                   (job->gpu-model-requested job-gpu-count-requested job (cached-queries/job->pool-name job)))]
     (->gpu-host-constraint job-gpu-count-requested job-gpu-model-requested)))
 
+; Cook uses mebibytes (MiB) for disk request and limit.
+; Convert bytes from k8s to MiB when passing to disk constraint,
+; and MiB back to bytes when submitting to k8s.
+(def disk-multiplier (* 1024 1024))
+
+(defrecord disk-host-constraint [job-disk-request job-disk-type]
+  JobConstraint
+  (job-constraint-name [this] (get-class-name this))
+  (job-constraint-evaluate
+    [this _ vm-attributes]
+    (job-constraint-evaluate this nil vm-attributes []))
+  (job-constraint-evaluate
+    [_ _ vm-attributes _]
+    (let [k8s-vm? (= (get vm-attributes "compute-cluster-type") "kubernetes")]
+      (if k8s-vm?
+        (let [vm-disk-type->space-available (get vm-attributes "disk")
+              ; k8s VMs all support disk, so they will only support jobs whose users requested disk or consume a default disk request
+              vm-satisfies-constraint? (if job-disk-request
+                                         (>= (get vm-disk-type->space-available job-disk-type 0) job-disk-request)
+                                         ; if job does not have a disk-request, pass true to always ignore the disk-host-constraint
+                                         true)]
+          [vm-satisfies-constraint? (when-not vm-satisfies-constraint?
+                                      "VM does not have enough disk space of requested disk type")])
+        ; Mesos jobs cannot request disk. If VM is a mesos VM, constraint always passes
+        (let [vm-satisfies-constraint? true]
+          [vm-satisfies-constraint?])))))
```

Comment:
Is the variable needed here? Seems superfluous

Reply:
yeah, will just return true then